pygmt/datasets/samples.py

```diff
@@ -360,6 +362,22 @@ def load_mars_shape(**kwargs):
     return data
 
 
+def _load_rock_sample_compositions():
+    """
+    Loads a table of rock sample compositions.
+
+    Returns
+    -------
+    data : pandas.DataFrame
+        The data table with columns "water", "air", "limestone", "z".
+    """
+
+    fname = which("@ternary.txt", download="c")
+    return pd.read_csv(
+        fname, sep="\t", header=None, names=["water", "air", "limestone", "z"]
+    )
```

Comment:
This `read_csv` call returns an incorrect table:
```
      water    air  limestone   z
0.16  0.331  0.509      9.344 NaN
0.22  0.625  0.158     39.156 NaN
0.09  0.180  0.732      2.621 NaN
0.66  0.258  0.078     12.062 NaN
0.72  0.154  0.124      7.961 NaN
...     ...    ...        ...  ..
0.08  0.222  0.699      3.340 NaN
0.83  0.071  0.102      6.371 NaN
0.77  0.058  0.177      5.574 NaN
0.48  0.100  0.419      4.254 NaN
0.50  0.339  0.161     15.203 NaN
```
We should use the following this:
```
pd.read_csv(fname, delim_whitespace=True, header=None, names=["water", "air", "limestone", "z"])
```
now the returned table is:
```
     water    air  limestone       z
0     0.16  0.331      0.509   9.344
1     0.22  0.625      0.158  39.156
2     0.09  0.180      0.732   2.621
3     0.66  0.258      0.078  12.062
4     0.72  0.154      0.124   7.961
..     ...    ...        ...     ...
995   0.08  0.222      0.699   3.340
996   0.83  0.071      0.102   6.371
997   0.77  0.058      0.177   5.574
998   0.48  0.100      0.419   4.254
999   0.50  0.339      0.161  15.203

[1000 rows x 4 columns]
```

Reply:
Following my comment https://github.com/GenericMappingTools/pygmt/pull/2211#discussion_r1038160762, we should use 
```
pd.read_csv(fname, delim_whitespace=True, header=None, names=["water", "air", "limestone"], usecols=(0, 1, 2))
```